test/e2e/storage/vsphere/config.go

```diff
@@ -0,0 +1,245 @@
+/*
+Copyright 2018 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package vsphere
+
+import (
+	"errors"
+	"fmt"
+	"gopkg.in/gcfg.v1"
+	"io"
+	"k8s.io/kubernetes/test/e2e/framework"
+	"os"
+)
+
+const (
+	vSphereConfFileEnvVar = "VSPHERE_CONF_FILE"
+)
+
+var (
+	confFileLocation = os.Getenv(vSphereConfFileEnvVar)
+)
+
+// Config represents vSphere configuration
+type Config struct {
+	Username          string
+	Password          string
+	Hostname          string
+	Port              string
+	Datacenters       string
+	RoundTripperCount uint
+}
+
+// ConfigFile represents the content of vsphere.conf file.
+// Users specify the configuration of one or more vSphere instances in vsphere.conf where
+// the Kubernetes master and worker nodes are running.
+type ConfigFile struct {
+	Global struct {
+		// vCenter username.
+		User string `gcfg:"user"`
+		// vCenter password in clear text.
+		Password string `gcfg:"password"`
+		// Deprecated. Use VirtualCenter to specify multiple vCenter Servers.
+		// vCenter IP.
+		VCenterIP string `gcfg:"server"`
```

Comment:
We should remove deprecated fields : VCenterIP, Datacenter, DefaultDatastore, WorkingDir, VMUUID and VMName

Reply:
Done. Tested with new changes.